src/test/java/org/openmrs/contrib/qaframework/helper/TestBase.java

```diff
@@ -103,13 +103,19 @@ public void failed(Throwable t, Description test) {
 
 	public TestBase() {
 		TestProperties testProperties = TestProperties.instance();
-		String sauceLabsUsername = testProperties.getProperty("SAUCELABS_USERNAME", null);
-		String sauceLabsAccessKey = testProperties.getProperty("SAUCELABS_ACCESSKEY", null);
-		sauceLabsHubUrl = testProperties.getProperty("saucelabs.hub.url","ondemand.saucelabs.com:80");
-
-		if (!StringUtils.isBlank(sauceLabsUsername)&& !StringUtils.isBlank(sauceLabsAccessKey)) {
-			sauceLabsAuthentication = new SauceOnDemandAuthentication(sauceLabsUsername, sauceLabsAccessKey);
-			sauceLabsResultReportingTestWatcher = new SauceOnDemandTestWatcher(this, sauceLabsAuthentication);
+		String sauceLabsUsername = testProperties.getProperty(
+				"SAUCELABS_USERNAME", null);
+		String sauceLabsAccessKey = testProperties.getProperty(
+				"SAUCELABS_ACCESSKEY", null);
```

Comment:
Yes @kdaud  as you can see this is not the best we can let go, because am still wondering if someone reviews this code and it looks like this, it would not show quality code. i don't know whether you understand me. Is it fine if we can try out another option of making our code look more cleaner than this, i believe we can find out another option of making our code look more proffessional.What is your suggestion.

Reply:
>Is it fine if we can try out another option of making our code look more cleaner than this, i believe we can find out another option of making our code look more proffessional

It may require some investigation on how this can be achieved of which am not have time for now.

>i don't know whether you understand me

yes i get point home.

Well, as of now the contributors are facing a challenge of the module being formatted on any local build plan and makes existing files tracked as changed files and have also realized some of them are not committing their changes as a result of the issue.

@sherrif10 if you have time, you can get the best solution to the issue otherwise I had thought this to be a temporarily solution to relieve the contributors pain as will get time in the near by future and investigate to find out the solution.

Reply:
Thanks @kdaud  for understanding the point, we need clean code. Ok let me investigate hope to come up with something, Meanwhile lets have final discussion of writing automated test in new bdd approach from distro.